contracts/NodeOperatorRegistry.sol

```diff
@@ -0,0 +1,642 @@
+    function updateSigner(bytes memory _signerPubkey)
+        external
+        override
+        whenNotPaused
+    {
+        uint256 operatorId = operatorOwners[msg.sender];
+        require(operatorId != 0, "Operator not exists");
+
+        Operator.NodeOperator storage no = operators[operatorId];
+
+        require(
+            no.status == Operator.NodeOperatorStatus.STAKED,
+            "Operator status not STAKED"
+        );
+
+        IValidator(no.validatorContract).updateSigner(
+            no.validatorId,
+            _signerPubkey
+        );
+
+        no.signerPubkey = _signerPubkey;
+        emit UpdateSignerPubkey(operatorId, no.validatorId, _signerPubkey);
+    }
+
+    /// @notice Allows withdraw heimdall fees
+    /// @param _accumFeeAmount accumulated heimdall fees
+    /// @param _index index
+    /// @param _proof proof
+    function claimFee(
+        uint256 _accumFeeAmount,
+        uint256 _index,
+        bytes memory _proof
+    ) external override whenNotPaused {
+        uint256 operatorId = operatorOwners[msg.sender];
+        require(operatorId != 0, "Operator not exists");
+
+        Operator.NodeOperator memory no = operators[operatorId];
+
+        require(
+            no.status == Operator.NodeOperatorStatus.EXIT,
+            "Operator status not EXIT"
+        );
+
+        require(_proof.length != 0, "Empty proof");
+        require(_accumFeeAmount != 0, "AccumFeeAmount is ZERO");
+        require(_index != 0, "index is ZERO");
+
+        IValidator(no.validatorContract).claimFee(
+            _accumFeeAmount,
+            _index,
+            _proof
+        );
+
+        emit ClaimFee(
+            operatorId,
+            no.validatorId,
+            _accumFeeAmount,
+            _index,
+            _proof
+        );
+    }
+
+    /// @notice Allows to update commission rate
+    /// @param _newCommissionRate new commission rate
+    function updateOperatorCommissionRate(
+        uint256 _operatorId,
+        uint256 _newCommissionRate
+    ) public override whenNotPaused userHasRole(UPDATE_COMMISSION_RATE_ROLE) {
+        Operator.NodeOperator memory no = operators[_operatorId];
+        if (no.status != Operator.NodeOperatorStatus.STAKED)
+            revert("Operator status no STAKED");
+
+        IValidator(no.validatorContract).updateCommissionRate(
+            no.validatorId,
+            _newCommissionRate
+        );
+
+        no.commissionRate = state.commissionRate;
+
+        emit UpdateCommissionRate(no.validatorId, _newCommissionRate);
+    }
+
+    /// @notice Allows to unjail the validator and turn his status from UNSTAKED to STAKED.
+    function unjail() external override whenNotPaused {
+        require(state.unjail, "Unjail is disabled");
+        
+        uint256 operatorId = operatorOwners[msg.sender];
+        require(operatorId != 0, "Operator not exists");
+
+        Operator.NodeOperator memory no = operators[operatorId];
```

Comment:
```suggestion
        Operator.NodeOperator storage no = operators[operatorId];
```
I believe that this structure wont be updated unless you declare it as a storage. If you declare something as a storage it actually takes a reference to it instead of just a value (memory). Line 554 won't have any effect if it isn't storage. You could also leave it as a memory but then you should do ```operators[operatorId] = no``` at the end.

Reply:
Yes, that right, it should be storage not memory